cmd/snap/cmd_run.go

```diff
@@ -403,6 +403,14 @@ func createUserDataDirs(info *snap.Info, opts *dirs.SnapDirOptions) error {
 	instanceUserData := info.UserDataDir(usr.HomeDir, opts)
 	instanceCommonUserData := info.UserCommonDataDir(usr.HomeDir, opts)
 	createDirs := []string{instanceUserData, instanceCommonUserData}
+
+	if opts.HiddenSnapDataDir {
```

Comment:
we shouldn't do this if the base is not core22+, no?

Reply:
I would have expected a new and seperate flag UseExposedUserSnapDir in place or something, that can be set only for snaps with core22+ bases, though it would always be the case that if that is true then HiddenSnapDataDir is true as well